src/test/javascript/spec/component/virtual-scroll/virtual-scroll.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { DebugElement, SimpleChange, SimpleChanges } from '@angular/core';
+import { ArtemisTestModule } from '../../test.module';
+import { VirtualScrollComponent } from 'app/shared/virtual-scroll/virtual-scroll.component';
+import { metisCoursePosts, metisCoursePostsWithCourseWideContext } from '../../helpers/sample/metis-sample-data';
+import { ReplaySubject } from 'rxjs';
+import { NavigationStart, Router, RouterEvent } from '@angular/router';
+import { Post } from 'app/entities/metis/post.model';
+
+const routerEventSubject = new ReplaySubject<RouterEvent>(1);
+
+class MockRouter {
+    public url = 'courses/1/discussion';
+    public events = routerEventSubject.asObservable();
+}
+
+describe('VirtualScrollComponent', () => {
+    let comp: VirtualScrollComponent<Post>;
+    let fixture: ComponentFixture<VirtualScrollComponent<Post>>;
+    let debugElement: DebugElement;
+
+    let prepareDataItemsSpy: jest.SpyInstance;
+    let forceReloadChangeSpy: jest.SpyInstance;
+
+    const minPostItemHeight = 126.7;
+
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [VirtualScrollComponent],
+            providers: [{ provide: Router, useClass: MockRouter }],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(VirtualScrollComponent);
+                comp = fixture.componentInstance;
+                debugElement = fixture.debugElement;
+                prepareDataItemsSpy = jest.spyOn(comp, 'prepareDataItems');
+                forceReloadChangeSpy = jest.spyOn(comp.forceReloadChange, 'emit');
+            });
+    });
+
+    it('should initialize correctly', fakeAsync(() => {
+        fixture.detectChanges();
+
+        expect(comp.prevOriginalItems).toBeEmpty();
+        expect(comp.minItemHeight).not.toBeNull();
+        expect(comp.focusInListener).not.toBeNull();
+        expect(comp.scrollListener).not.toBeNull();
+    }));
+
+    it('should set originalItems to empty array if undefined value is passed into the component', fakeAsync(() => {
+        fixture.detectChanges();
+
+        const changes = {} as SimpleChanges;
+        changes.originalItems = new SimpleChange([], undefined, true);
+        comp.originalItems = undefined;
+        comp.ngOnChanges(changes);
+        tick();
+        fixture.detectChanges();
+
+        expect(comp.originalItems).toBeDefined();
+        expect(comp.originalItems).toBeEmpty();
+    }));
+
+    it('should initialize DOM tree with items', fakeAsync(() => {
+        fixture.detectChanges();
+        comp.forceReload = true;
+        comp.originalItems = metisCoursePosts;
```

Comment:
You should replace this with dummy items and not only posts as this is a generic component

Reply:
I don't understand what is wrong with testing using a real-case scenario here. The dummy classes would have to have an id and some variables anyways.

Reply:
Because you are suddenly making a generic standalone component depend on the test to posts. It is kind of confusing. But if you do not want to change it ok